lib/Runtime/Library/WebAssemblyInstance.cpp

```diff
@@ -260,257 +284,198 @@ void WebAssemblyInstance::BuildObject(WebAssemblyModule * wasmModule, ScriptCont
+                    JavascriptError::ThrowTypeError(ctx, WASMERR_NeedMemoryObject);
+                }
+                WebAssemblyMemory * mem = WebAssemblyMemory::FromVar(prop);
+                if (!wasmModule->IsValidMemoryImport(mem))
+                {
+                    JavascriptError::ThrowTypeError(ctx, WASMERR_NeedMemoryObject);
+                }
+                env->SetMemory(counter, mem);
+            }
             break;
         }
-        case Wasm::WasmTypes::F32:
+        case Wasm::ExternalKinds::Table:
         {
-            float val = (float)JavascriptConversion::ToNumber(prop, ctx);
-            global->cnst.f32 = val;
-            SetGlobalValue(moduleEnv, offset, val);
+            Assert(wasmModule->HasTableImport());
+            if (wasmModule->HasTableImport())
+            {
+                if (!WebAssemblyTable::Is(prop))
+                {
+                    JavascriptError::ThrowTypeError(ctx, WASMERR_NeedTableObject);
+                }
+                WebAssemblyTable * table = WebAssemblyTable::FromVar(prop);
+
+                if (!wasmModule->IsValidTableImport(table))
+                {
+                    JavascriptError::ThrowTypeError(ctx, WASMERR_NeedTableObject);
+                }
+                env->SetTable(counter, table);
+            }
             break;
         }
-        case Wasm::WasmTypes::F64:
+        case Wasm::ExternalKinds::Global:
         {
-            double val = JavascriptConversion::ToNumber(prop, ctx);
-            global->cnst.f64 = val;
-            SetGlobalValue(moduleEnv, offset, val);
+            Wasm::WasmGlobal* global = wasmModule->GetGlobal(counter);
+            if (global->IsMutable() || (!JavascriptNumber::Is(prop) && !TaggedInt::Is(prop)))
+            {
+                JavascriptError::ThrowTypeError(ctx, WASMERR_InvalidImport);
+            }
+
+            Assert(global->GetReferenceType() == Wasm::GlobalReferenceTypes::ImportedReference);
+            Wasm::WasmConstLitNode cnst = {0};
+            switch (global->GetType())
+            {
+            case Wasm::WasmTypes::I32: cnst.i32 = JavascriptConversion::ToInt32(prop, ctx); break;
+            case Wasm::WasmTypes::F32: cnst.f32 = (float)JavascriptConversion::ToNumber(prop, ctx); break;
+            case Wasm::WasmTypes::F64: cnst.f64 = JavascriptConversion::ToNumber(prop, ctx); break;
+            case Wasm::WasmTypes::I64: Js::JavascriptError::ThrowTypeError(ctx, WASMERR_InvalidTypeConversion);
+            default:
+                Assert(UNREACHED);
+                JavascriptError::ThrowTypeError(ctx, VBSERR_InternalError);
+            }
+            env->SetGlobalValue(global, cnst);
             break;
         }
-        case Wasm::WasmTypes::I64:
-            Js::JavascriptError::ThrowTypeError(ctx, WASMERR_InvalidTypeConversion);
         default:
             Assert(UNREACHED);
-            break;
-
+            JavascriptError::ThrowTypeError(ctx, VBSERR_InternalError);
```

Comment:
You can do Js::ThrowInternalError(), which will assert on chk build and throw error on test/release build